lib/search/service.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+require 'common/client/base'
+require 'search/response'
+
+module Search
+  class Service < Common::Client::Base
+    include Common::Client::Monitoring
+
+    STATSD_KEY_PREFIX = 'api.search'
+
+    configuration Search::Configuration
+
+    attr_reader :query
+
+    def initialize(query)
+      @query = query
+    end
+
+    # GETs a list of search results from Search.gov web results API
+    # @return [Search::ResultsResponse] wrapper around results data
+    #
+    def results
+      with_monitoring do
+        response = perform(:get, results_url, query_params)
+        Search::ResultsResponse.from(response)
+      end
+    rescue StandardError => error
+      handle_error(error)
+    end
+
+    private
+
+    def results_url
+      config.base_path
+    end
+
+    # Required params [affiliate, access_key, query]
+    # Optional params [enable_highlighting, limit, offset, sort_by]
+    #
+    # @see https://search.usa.gov/sites/7378/api_instructions
+    #
+    def query_params
+      {
+        affiliate:  affiliate,
+        access_key: access_key,
+        query:      query
+      }
+    end
+
+    def affiliate
+      Settings.search.affiliate
+    end
+
+    def access_key
+      Settings.search.access_key
+    end
+
+    def handle_error(error)
+      case error
+      when Common::Client::Errors::ClientError
+        message = parse_messages(error).first
+        log_error_message(message)
+        # raise_backend_exception('SEARCH_400', self.class, error) if message.present?
```

Comment:
@hpjaj do you think we want to raise a backend exception for a given `400` level status? Or just pass through the message to the response class? 

I'm trying to coerce the search.gov api into giving me some 500 level error codes but have had no luck, and as we know their docs have zero examples... but those are the codes I'd imagine want an exception raised. Thoughts?